### PR TITLE
Oriented overset updates

### PR DESCRIPTION
**Pull request by wjhorne**
Includes : 

- Removal of TIOGA TPL
- Upgrade to STK assembly including oriented shape cutting + documentation
- Bug fixes for 6-DOF motion 
- Name shift from orphan terminology to constraint terminology.

### CODE REVIEW

**Comment by spdomin** [comment]
@wjhorne, this looks good. However, before we squash and merge, let's work through the Trilinos diffs/bi-sect first.

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 breathe==4.13.1
 docutils==0.14
-Pygments==2.7.4
+Pygments==2.3.1
```
Comment (spdomin) [inline docs/requirements.txt]: I think that this needs to be 2.7.4. Not sure how the merge did not deal with this automatically?

File: src/AssembleMomentumElemSolverAlgorithm.C [hunk 2]
```diff
@@ -369,7 +369,8 @@ AssembleMomentumElemSolverAlgorithm::execute()
         for (int i = 0; i < nDim; ++i ) {
           // udotx
           const double dxi = p_coordinates[irNdim+i]-p_coordinates[ilNdim+i];
-          const double ui = 0.5*(p_vrtm[ilNdim+i] + p_vrtm[irNdim+i]);
+          //const double ui = 0.5*(p_vrtm[ilNdim+i] + p_vrtm[irNdim+i]);
```
Comment (spdomin) [inline src/AssembleMomentumElemSolverAlgorithm.C]: Try to remove old comments..

File: src/Realm.C [hunk 3]
```diff
@@ -2904,7 +2935,7 @@ Realm::set_mesh_velocity_six_dof(
       mesh_velocity_cross_product(motion.bodyOmega_.data(), cX, uX);
       
       for ( int i = 0; i < nDim; ++i ) {
-        vnp1[offSet+i] =  uX[i]+motion.bodyVel_[i];
+        vnp1[offSet+i] = (uX[i]+motion.bodyVel_[i]);
```
Comment (spdomin) [inline src/Realm.C:2938]: Might only be style...

File: src/kernel/VolumeOfFluidSucvNsoElemKernel.C [hunk 4]
```diff
@@ -140,6 +151,7 @@ VolumeOfFluidSucvNsoElemKernel<AlgTraits>::execute(
       const DoubleType vofIc = v_vofNp1(ic);
       for ( int j = 0; j < AlgTraits::nDim_; ++j ) {
         w_uIp[j] += r*v_vrtm(ic,j);
+        w_uIpv[j] += r*v_v(ic,j);
```
Comment (spdomin) [inline src/kernel/VolumeOfFluidSucvNsoElemKernel.C]: Not needed now?

File: src/kernel/VolumeOfFluidSucvNsoElemKernel.C [hunk 5]
```diff
@@ -99,6 +107,7 @@ VolumeOfFluidSucvNsoElemKernel<AlgTraits>::execute(
   ScratchViews<DoubleType>& scratchViews)
 {
   NALU_ALIGNED DoubleType w_uIp[AlgTraits::nDim_];
+  NALU_ALIGNED DoubleType w_uIpv[AlgTraits::nDim_];
```
Comment (spdomin) [inline src/kernel/VolumeOfFluidSucvNsoElemKernel.C:104]: Not needed now?

File: src/kernel/VolumeOfFluidSucvNsoElemKernel.C [hunk 6]
```diff
@@ -64,6 +68,10 @@ VolumeOfFluidSucvNsoElemKernel<AlgTraits>::VolumeOfFluidSucvNsoElemKernel(
   // fields and data
   dataPreReqs.add_coordinates_field(*coordinates_, AlgTraits::nDim_, CURRENT_COORDINATES);
   dataPreReqs.add_gathered_nodal_field(*velocityRTM_, AlgTraits::nDim_);
+
+  // NEW WJH
```
Comment (spdomin) [inline src/kernel/VolumeOfFluidSucvNsoElemKernel.C]: WIP... Probably can remove..

**Review by spdomin — CHANGES_REQUESTED** [review]
This looks good. I think you only have a few WIP comments/code snippets to remove. 

Once that is done, we can approve..

**Comment by wjhorne** [comment]
Sounds good. I'll clean this all up. Thank you for reviewing it

**Comment by wjhorne** [comment]
I found that many of these suggested changes were from the first commit out of the two that were initially in the PR. The second commit had already fixed many of the WIP issues. There was one remaining WIP line found that still needed to be removed and has now been removed. I believe all changes have now been accounted for.

**Comment by spdomin** [comment]
Looks good. Most times I squash my commits prior to a PR. In the end, I squash them anyway during the merge.

Next time, I will look more carefully.

**Pull request merged**